src/components/parameters-picker.tsx

```diff
@@ -18,39 +19,95 @@ export type ParametersPickerProps = {
   removeParameter: (idx: number) => void;
   // CSS classes for elements
   formPrefix: string;
+  errors: Scheduler.ErrorsType;
+  handleErrorsChange: (errors: Scheduler.ErrorsType) => void;
+};
+
+const useForceUpdate = () => {
+  const [, setTick] = useState(0);
+  const update = useCallback(() => {
+    setTick(tick => tick + 1);
+  }, []);
+  return update;
 };
```

Comment:
Why is this required?

Reply:
Without `useForceUpdate`, although the model is updated, the parameters component does not automatically rerender. This function is meant to update a nominal state to force a rerender.

Reply:
Do we know why this behavior is happening? 
That seems like a hack to support a refresh that React supports by design; this is likely an issue with our code. If `ParametersPicker` is a functional component, any update to props is supposed to re-render it.  If you need internal state in the `ParametersPicker` to trigger changes, then that's the way to go. In either case, there shouldn't be a pattern of adding an unused state to force updates.

Reply:
> A parameter with a name and an empty-string value is not inherently an error; there might be a reasonable reason for that. (The `errors` variable uses an empty-string value to mean "no error" for a particular input name, for example.) @ellisonbg had suggested the specific case of "empty parameter name, non-empty parameter value" being an error, but I don't consider the other three cases to be problematic:
> 
> Empty name	Name
> **Empty value**	✅	✅
> **Value**	❌	✅

From a user's perspective, there is a very low chance they would want to pass an empty string for a parameter; for other cases (number, boolean) this will cause an exception. On the other hand, blocking users to pass an empty value will hinder some users, I don't feel that we will have a lot of users who would want this. Ok to submit this as-is and revisit after some user feedback.

Reply:
BTW, the async update was fixed in #63.